example/src/components/RoomParticipant.tsx

```diff
@@ -24,40 +42,93 @@ export const RoomParticipant = ({
     return false;
   };
 
-  return (
-    <View>
-      {participantHasVideo() ? (
-        <Membrane.VideoRendererView
-          trackId={videoTrack!.id}
-          style={styles.videoTrack}
-        />
-      ) : (
-        <View style={styles.videoTrack}>
-          <NoCameraView username={metadata.displayName} />
-        </View>
-      )}
-      <View style={styles.displayNameContainer}>
-        <View
-          style={[
-            styles.displayName,
-            type === 'Local' ? styles.localUser : styles.remoteUser,
-          ]}
-        >
-          <Typo variant="label" color={TextColors.white}>
-            {type === 'Local' ? 'You' : metadata.displayName}
-          </Typo>
-        </View>
-      </View>
+  const getTextForPinButton = () => {
+    return focused ? 'Unpin user' : tileSmall ? 'Pin' : 'Pin user';
+  };
+
+  const onPinButton = () => {
+    if (focused) {
+      onPinButtonPressed(null);
+      return;
+    }
+    onPinButtonPressed(id);
+  };
 
-      {!audioTrack?.metadata.active && (
-        <View style={styles.mutedIcon}>
-          <Icon
-            name="Microphone-off"
-            size={16}
-            color={BrandColors.darkBlue100}
+  const opacityStyle = useAnimatedStyle(() => {
+    return {
+      opacity: buttonOpacity.value,
+    };
+  });
+
+  const triggerShowingPinButton = async () => {
+    if (pinButtonHiddden || isPinButtonShown.current) {
+      return;
+    }
+
+    buttonOpacity.value = withTiming(1, { duration: 300 });
+    isPinButtonShown.current = true;
+    setShowPinButton(true);
+    await delay(1700);
+    buttonOpacity.value = withTiming(0, { duration: 300 });
```

Comment:
the `withTiming` function has a callback argument https://docs.swmansion.com/react-native-reanimated/docs/api/animations/withTiming#callback-functionoptional

and you have also `withDelay` and `withSequence` functions but I'm not sure if they'll be useful here
(maybe you can make a sequence of timing, delay and timing animations)

Reply:
Added proper sequence with timings.